jobs/pacman-cloud-discovery/src/main/java/com/tmobile/cso/pacman/inventory/util/InventoryUtil.java

```diff
@@ -1993,17 +1993,26 @@ public static Map<String, List<SnapshotVH>> fetchSnapshots(BasicSessionCredentia
      * Get SnapShot permission.
      *
      * @param ec2Client  the AmazonEC2
-     * @param snapshotID the SnapShot ID
+     * @param snapshot the SnapShot
      * @return the boolean
      */
-    private static boolean getSnapShotPermissions(AmazonEC2 ec2Client, String snapshotID) {
-        DescribeSnapshotAttributeResult describeSnapshotAttribute = ec2Client
-                .describeSnapshotAttribute(new DescribeSnapshotAttributeRequest().withSnapshotId(snapshotID)
-                        .withAttribute(SnapshotAttributeName.CreateVolumePermission));
-        List<CreateVolumePermission> createVolumePermissions = describeSnapshotAttribute.getCreateVolumePermissions();
-        boolean ispublic = createVolumePermissions.stream()
-                .anyMatch(volPerm -> volPerm.getGroup() != null && "all".equalsIgnoreCase(volPerm.getGroup()));
-        return ispublic;
+    private static SnapshotVH buildSnapshotVH(AmazonEC2 ec2Client, Snapshot snapshot) {
+        try {
+            DescribeSnapshotAttributeResult describeSnapshotAttribute = ec2Client
+                    .describeSnapshotAttribute(new DescribeSnapshotAttributeRequest().withSnapshotId(snapshot.getSnapshotId())
+                            .withAttribute(SnapshotAttributeName.CreateVolumePermission));
+            List<CreateVolumePermission> createVolumePermissions = describeSnapshotAttribute.getCreateVolumePermissions();
+            boolean isPublic = createVolumePermissions.stream()
+                    .anyMatch(volPerm -> volPerm.getGroup() != null && "all".equalsIgnoreCase(volPerm.getGroup()));
+            return new SnapshotVH(snapshot, isPublic);
```

Comment:
@ershad-paladin @santhosh-challa do we need to handle the other permissions like "explicit" or "implicit"? Do we only care about if its public for a specific policy?
![image](https://github.com/PaladinCloud/CE/assets/31665420/dd92ceb9-516f-4271-816e-3da1a5727785)